Newsletters: add 147 (2021-05-05) (#566)
bitcoinops/bitcoinops.github.io: _posts/en/newsletters/2021-05-05-newsletter.md

    period (assuming there's no reorg):

    ```text
    $ bitcoin-cli getblockchaininfo \
      | jq '.softforks.taproot.bip9.statistics | .elapsed,.count,.possible'
    353
    57
    false
    ```

    If you prefer a graphical representation with supplementary
    information about miner progress and don't need to use your own
    node, we recommend [Taproot.watch][] by Hampus Sjöberg.

## News

- **Closing lost channels with only a BIP32 seed:** as described in [Newsletter
  #128][news128 ln ecdh], Lloyd Fournier proposed a method for creating
  new channels that would allow a user who lost all information except
  for their BIP32 wallet seed to re-discover their peers using only
  public information about the LN network.  Once the user found their
  peers, they could request the peers close their mutual channels using
  the [BOLT2][] data loss protection protocol (see [Newsletter
  #31][news31 data_loss]).  The proposed method isn't perfect---users
  should still create backups[^missing-peer] and replicate them across
  independent systems---but Fournier's proposal provides additional
  redundancy that would be especially useful for everyday users.

>     A couple weeks ago, Rusty Russell restarted the [thread][russell

Reviewer: I know that "a couple weeks" can be used in American English, but I don't know whether it's considered informal or not (in British English we'd always use "a couple of weeks"). I see a mixture of "a couple {things}" and "a couple of {things}" in previous newsletters.

    ecdh channels] after trying to [specify][russell ecdh spec] and
    implement the idea.  After some additional mailing list discussion with Fournier
    and a group [conversation][lndev deterministic] in the weekly LN
    protocol development meeting, Russell indicated he was leaning
    against the idea, [saying][russell backups] "I see encrypted backups
    as a more-likely-to-be-implemented solution though.  Because they're
    useful to send to places other than peers, and they can also contain
    [HTLC][topic htlc] information if you want."  Being able to contain

>     [HTLC][topic htlc] information would allow settling payments that

Reviewer: No need to link to [topic htlc] from both of these. Perhaps remove the link from the Russell quote.

Author: Whoops, thanks.  (I added the link after pasting in Rusty's quote, then wrote the subsequent line and thought the link would be better there, then forgot to remove the old link.)

    mechanism based solely on a BIP32 seed could provide.

## Releases and release candidates

*New releases and release candidates for popular Bitcoin infrastructure
projects.  Please consider upgrading to new releases or helping to test
release candidates.*

- [Bitcoin Core 0.21.1][Bitcoin Core 0.21.1] is a new version of Bitcoin
  Core that contains activation logic for the proposed [taproot][topic
  taproot] soft fork.  Taproot uses [schnorr signatures][topic schnorr
  signatures] and allows the use of [tapscript][topic tapscript].  These
  are, respectively, specified by BIPs [341][BIP341], [340][BIP340], and
  [342][BIP342].  Also included is the ability to pay [bech32m][topic
  bech32] addresses specified by [BIP350][], although bitcoins spent to
  such addresses on mainnet will not be secure until activation of a
  soft fork using such addresses, such as taproot.  The release
  additionally includes bug fixes and minor improvements.

    Note: due to a [problem][wincodesign] with the certificate authority
    that provides the code signing certificates for the Windows versions
    of Bitcoin Core, users on Windows will need to click through an

>     extra prompt to install.  It is also expected that there will be a

Reviewer: ```suggestion
    extra prompt to install.  It is expected that there will be a
```

    0.21.1.1 release with an updated certificate when the problem is
    fixed.  If you are planning to upgrade anyway, there's no reason to
    delay using 0.21.1 because of this problem.

- [BTCPay 1.1.0][] is the latest major release for this self-hosted payment
  processing software. It includes [Lightning Loop][news53 lightning loop]
  support, adds [WebAuthN/FIDO2][fido2 website] as a two-factor authentication
  option, makes various UI improvements, and marks a switch to using [semantic
  versioning][semantic versioning website] for version numbers moving forward.

## Notable code and documentation changes

*Notable changes this week in [Bitcoin Core][bitcoin core repo],
[C-Lightning][c-lightning repo], [Eclair][eclair repo], [LND][lnd repo],
[Rust-Lightning][rust-lightning repo], [libsecp256k1][libsecp256k1
repo], [Hardware Wallet Interface (HWI)][hwi repo],
[Rust Bitcoin][rust bitcoin repo], [BTCPay Server][btcpay server repo],
[Bitcoin Improvement Proposals (BIPs)][bips repo], and [Lightning
BOLTs][bolts repo].*

- [Bitcoin Core #19160][] is the next PR in the [multiprocess project][bitcoin
  core multiprocess] and adds the ability for the `bitoin-node` process to spawn

>   other processes and communicate with them using [Capn' Proto][capn proto].

Reviewer 2: ```suggestion
  other processes and communicate with them using [Cap’n Proto][capn proto].
```
(noticed while opening the file; haven't read the rest yet and will try to do a full read-through a bit later)

  versioning][semantic versioning website] for version numbers moving forward.

## Notable code and documentation changes

*Notable changes this week in [Bitcoin Core][bitcoin core repo],
[C-Lightning][c-lightning repo], [Eclair][eclair repo], [LND][lnd repo],
[Rust-Lightning][rust-lightning repo], [libsecp256k1][libsecp256k1
repo], [Hardware Wallet Interface (HWI)][hwi repo],
[Rust Bitcoin][rust bitcoin repo], [BTCPay Server][btcpay server repo],
[Bitcoin Improvement Proposals (BIPs)][bips repo], and [Lightning
BOLTs][bolts repo].*

- [Bitcoin Core #19160][] is the next PR in the [multiprocess project][bitcoin
  core multiprocess] and adds the ability for the `bitoin-node` process to spawn
  other processes and communicate with them using [Capn' Proto][capn proto].
  These features are currently only used for testing, but the [next PR][Bitcoin
  Core #10102] in the project will allow Bitcoin Core to be started in
  multiprocess mode with the `bitcoin-core` process spawning separate
  `bitcoin-wallet` and `bitcoin-gui` processes.

- [Bitcoin Core #21009][] removes the RewindBlockIndex logic triggered when

>   updating a pre-segwit client (v0.13.0 or older) to a segwit-enforcing

Author: @Xekyo FYI, I'm going to s/client/node/ throughout this paragraph.  I prefer to use "client" for tools that are dependent on other tools, e.g. lightweight clients.  I use "node" or "peer" for tools that are full participants in the network.

    extra prompt to install.  It is expected that there will be a
    0.21.1.1 release with an updated certificate when the problem is
    fixed.  If you are planning to upgrade anyway, there's no reason to
    delay using 0.21.1 because of this problem.

- [BTCPay 1.1.0][] is the latest major release for this self-hosted payment
  processing software. It includes [Lightning Loop][news53 lightning loop]
  support, adds [WebAuthN/FIDO2][fido2 website] as a two-factor authentication
  option, makes various UI improvements, and marks a switch to using [semantic
  versioning][semantic versioning website] for version numbers moving forward.

## Notable code and documentation changes

*Notable changes this week in [Bitcoin Core][bitcoin core repo],
[C-Lightning][c-lightning repo], [Eclair][eclair repo], [LND][lnd repo],
[Rust-Lightning][rust-lightning repo], [libsecp256k1][libsecp256k1
repo], [Hardware Wallet Interface (HWI)][hwi repo],
[Rust Bitcoin][rust bitcoin repo], [BTCPay Server][btcpay server repo],
[Bitcoin Improvement Proposals (BIPs)][bips repo], and [Lightning
BOLTs][bolts repo].*

- [Bitcoin Core #19160][] is the next PR in the [multiprocess project][bitcoin

>   core multiprocess] and adds the ability for the `bitoin-node` process to spawn

Reviewer 3: ```suggestion
  core multiprocess] and adds the ability for the `bitcoin-node` process to spawn
```

  it is synced, `gettxoutsetinfo` runs nearly instantly and users can specify a
  height or block hash for the statistics.  Being able to get the
  statistics for a particular block will be especially useful for
  allowing community verification of [assumeUTXO][topic assumeutxo]
  chainstate archives.

- [Bitcoin Core #21009][] removes the RewindBlockIndex logic triggered when
  updating a pre-segwit node (v0.13.0 or older) to a segwit-enforcing
  version. Pre-segwit nodes only processed stripped blocks that lacked the (segregated) witness
  data. The RewindBlockIndex logic discarded its copies of such blocks,
  re-downloaded them in their complete form, and validated them using segwit's rules. As pre-segwit
  nodes have been end-of-life since 2018, the scenario has become uncommon.
  Future releases will instead prompt the user to reindex for an equivalent
  outcome.

- [LND #5159][] builds on [previous work][news144 lnd ampsendpayment] to add
  support for making spontaneous Atomic Multipath Payments (AMPs) by manually
  specifying payment parameters to the `sendpayment` RPC. Invoking `sendpayment`

>   with an AMP invoice is expected to be implemented in followup PRs.

Reviewer 2: s/followup/follow-up/

> - [Rust-Lightning #893][] only allows accepting a payment if it included

Reviewer 2: s/included/includes/

  a payment secret.  Payment secrets are created by the receiver and
  included in invoices.  The spender includes the payment secret in
  their payment in order to prevent attempts by third parties to reduce
  the privacy of [multipath payments][topic multipath payments].  Alongside
  this change are several API changes designed to reduce the chance that
  a payment will be accepted incorrectly.

- [BIPs #1104][] updates the [BIP341][] specification of [taproot][topic
  taproot] with activation parameters based on the Speedy Trial proposal
  (see [Newsletter #139][news139 speedy trial]).

## Footnotes

[^missing-peer]:
    The data loss protection protocol, and other proposed methods such

>     as [covert requests for mutual channel closes][news128 covert] requires your channel peer to

Reviewer 2: - add comma or alternatively, s/requires/require/
- consider s/peer/peers/ as they are referred to as "they" in the next sentence

```suggestion
    as [covert requests for mutual channel closes][news128 covert], requires your channel peers to
```
